packages/curve-ui-kit/src/shared/ui/NumericTextField.tsx

```diff
@@ -0,0 +1,142 @@
+import { useEffect, useState } from 'react'
+import TextField from '@mui/material/TextField'
+import type { TextFieldProps } from '@mui/material/TextField'
+
+/**
+ * Validates and normalizes numeric input by replacing commas with dots
+ * and ensuring only valid numeric characters are allowed.
+ * Prevents multiple decimal points and ensures minus sign is only at the beginning.
+ *
+ * @param value - The new input value to validate
+ * @param current - The current input value to fall back to if validation fails
+ * @returns The validated and normalized input value, or the current value if invalid
+ */
+const sanitize = (value: string, current: string): string => {
```

Comment:
as you mentioned yourself, this kind of code is fragile and dependent on browsers and locales.
In general I'd say it's better to avoid such customization and rely on browser implementation

Reply:
Sadly I've never seen default browser behavior behave as desired for number inputs, for any job I've ever worked. I hoped maybe MUI would take care of it, but it doesn't look like they've done anything wrt the behavior, only the styling 😞